storage/rocksdb/rdb_io_watchdog.cc

```diff
@@ -0,0 +1,233 @@
+  // buffer cache.
+  int fd = open(fname.c_str(), O_WRONLY | O_DIRECT | O_CREAT | O_SYNC,
+                S_IRWXU | S_IWUSR);
+
+  if (unlikely(fd == -1)) {
+    return fd;
+  }
+
+  int ret = write(fd, m_buf, RDB_IO_WRITE_BUFFER_SIZE);
+
+  if (unlikely(ret != RDB_IO_WRITE_BUFFER_SIZE)) {
+    return ret;
+  }
+
+  ret = close(fd);
+
+  if (unlikely(ret)) {
+    return ret;
+  }
+
+  ret = unlink(fname.c_str());
+
+  if (unlikely(ret)) {
+    return ret;
+  }
+
+  return HA_EXIT_SUCCESS;
+}
+
+int Rdb_io_watchdog::reset_timeout(const uint32_t &write_timeout) {
+  // This function will be called either from a thread initializing MyRocks
+  // engine or handling system variable changes. We need to account for the
+  // possibility of I/O callback executing at the same time. If that happens
+  // then we'll wait for it to finish.
+  RDB_MUTEX_LOCK_CHECK(m_reset_mutex);
+
+  struct sigevent e;
+
+  // In all the cases all the active timers needs to be stopped.
+  int ret = stop_timers();
+
+  if (unlikely(ret)) {
+    // NO_LINT_DEBUG
+    sql_print_warning("Stopping I/O timers failed with %d.", errno);
+    RDB_MUTEX_UNLOCK_CHECK(m_reset_mutex);
+    return ret;
+  }
+
+  m_write_timeout = write_timeout;
+  m_io_in_progress.store(false);
+
+  // Zero means that the I/O timer will be disabled. Therefore there's nothing
+  // for us to do here.
+  if (!write_timeout) {
+    RDB_MUTEX_UNLOCK_CHECK(m_reset_mutex);
+    return HA_EXIT_SUCCESS;
+  }
+
+  free(m_buf);
+
+  ret = posix_memalign(reinterpret_cast<void **>(&m_buf),
+                       RDB_IO_WRITE_BUFFER_SIZE, RDB_IO_WRITE_BUFFER_SIZE);
+
+  if (unlikely(ret)) {
+    m_buf = nullptr;
+    RDB_MUTEX_UNLOCK_CHECK(m_reset_mutex);
+    // NB! The value of errno is not set.
+    return ret;
+  }
+
+  DBUG_ASSERT(m_buf != nullptr);
+  memset(m_buf, 0, RDB_IO_WRITE_BUFFER_SIZE);
+
+  // Common case gets handled here - we'll create a timer with a specific
+  // interval to check a set of directories for write access.
+  DBUG_ASSERT(m_dirs_to_check.size() > 0);
+
+  e.sigev_notify = SIGEV_THREAD;
+  e.sigev_notify_function = &Rdb_io_watchdog::io_check_callback_wrapper;
+  e.sigev_value.sival_ptr = this;
+  e.sigev_notify_attributes = nullptr;
+
+  ret = timer_create(CLOCK_MONOTONIC, &e, &m_io_check_timer);
+
+  if (unlikely(ret)) {
+    // NO_LINT_DEBUG
+    sql_print_warning("Creating a I/O timer failed with %d.", errno);
+    RDB_MUTEX_UNLOCK_CHECK(m_reset_mutex);
+    return ret;
+  }
+
+  struct itimerspec timer_spec;
+  memset(&timer_spec, 0, sizeof(timer_spec));
+
+  // I/O timer will need to execute on a certain interval.
+  timer_spec.it_value.tv_sec = m_write_timeout;
+  timer_spec.it_interval.tv_sec = m_write_timeout;
+
+  ret = timer_settime(m_io_check_timer, 0, &timer_spec, nullptr);
+
+  if (unlikely(ret)) {
+    // NO_LINT_DEBUG
+    sql_print_warning("Setting time for a watchdog I/O timer failed with %d.",
+                      errno);
+  }
```

Comment:
Looking at the use of this function, I see no reason why it can/should even return a value as the return from it is never checked nor any action taken. It seems that the error handling for it should always be to just sql_print_warning and return;